### PR TITLE
pr

### PR DESCRIPTION
**Pull request by AlexGuler**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
+        }
+    }
+}
+
+dependencies {
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
```
Comment (lpalac4) [inline app/build.gradle:28]: #nth The way the project automatically pulls out this $kotlin_version variable out of the build gradle of an individual project and lets you manage it in a separate file is smart.  We should consider doing the same with all library dependency versions.  That way you'll have one file to manage all versions.

File: .idea/misc.xml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (lpalac4) [inline .idea/misc.xml:1]: #nth We might want to add to the .gitignore any files from *.idea* here

File: app/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+/build
```
Comment (lpalac4) [inline app/.gitignore:1]: #nth Adding to .ignore would be good here too

Comment (grndvl1) [inline app/.gitignore:1]: add .idea, build and .gradle dirs to this file along with other files like local.properties...

File: app/src/main/res/layout/activity_emoji_puzzle.xml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        tools:context=".PuzzlePage.EmojiPuzzleActivity">
+
+    <TextView
+            android:id="@+id/categoryTitle"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            app:layout_constraintLeft_toLeftOf="parent"
+            app:layout_constraintRight_toRightOf="parent"
+            app:layout_constraintTop_toTopOf="parent"
+            android:text="Category"
```
Comment (lpalac4) [inline app/src/main/res/layout/activity_emoji_puzzle.xml:17]: #nth Hardcoded text should be moved to string resources for better optimization

File: app/src/main/res/layout/activity_main.xml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+<androidx.constraintlayout.widget.ConstraintLayout
+        xmlns:android="http://schemas.android.com/apk/res/android"
+        xmlns:tools="http://schemas.android.com/tools"
+        xmlns:app="http://schemas.android.com/apk/res-auto"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        tools:context=".MainPage.MainActivity">
+
+    <TextView
+            android:id="@+id/welcomeText"
```
Comment (lpalac4) [inline app/src/main/res/layout/activity_main.xml:11]: #nth consider a title style that can be shared, where you have 18sp and textview styling

File: app/src/main/res/layout/cell_choose_category.xml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+        xmlns:app="http://schemas.android.com/apk/res-auto"
+        android:orientation="vertical">
+
+    <Button
+            android:id="@+id/categoryButton"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:textSize="17sp"
+            android:textAllCaps="false"
+            android:text="Category"
```
Comment (lpalac4) [inline app/src/main/res/layout/cell_choose_category.xml:15]: #nth More hardcoded strings

File: app/src/main/java/com/example/emojigame/MainPage/MainActivity.kt [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        setupRecyclerView()
+
+        loadingProgressbar.show()
+        loadCategories()
+    }
+
+    private fun loadCategories() {
+        GlobalScope.launch(Dispatchers.Main) {
```
Comment (lpalac4) [inline app/src/main/java/com/example/emojigame/MainPage/MainActivity.kt:33]: #req Although GlobalScope can be useful for quick and dirty projects (like dis one) remember GlobalScope launches your coroutine in global scope outside of the context of your activity.  Imagine what would happen if your app was rotated before your global scope coroutine completes.  https://proandroiddev.com/kotlin-coroutines-patterns-anti-patterns-f9d12984c68e

File: app/src/main/java/com/example/emojigame/PuzzlePage/EmojiPuzzleActivity.kt [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+            GlobalScope.launch(Dispatchers.Main) {
+                loadingProgressbar.hide()
+                val puzzles = API.getPuzzlesOfCategory(category)
+                emojiPuzzleAdapter.submitList(puzzles)
+            }
+        } else {
+            loadingProgressbar.hide()
+            Toast.makeText(
+                this@EmojiPuzzleActivity,
+                "Error occurred could not retrieve puzzles, please try again later.",
```
Comment (lpalac4) [inline app/src/main/java/com/example/emojigame/PuzzlePage/EmojiPuzzleActivity.kt:41]: #nth lets move this hardcoded string to resources

File: app/src/main/java/com/example/emojigame/PuzzlePage/EmojiPuzzleActivity.kt [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+        categoryTitle.text = category
+
+        setupPuzzlesRecyclerView()
+        loadingProgressbar.show()
+        loadPuzzles(category)
+    }
+
+    private fun loadPuzzles(category: String?) {
+        if (category != null) {
+            GlobalScope.launch(Dispatchers.Main) {
```
Comment (lpalac4) [inline app/src/main/java/com/example/emojigame/PuzzlePage/EmojiPuzzleActivity.kt:32]: #req Same idea as the other GlobalScope coroutine being launched from an Activity

File: app/src/main/java/com/example/emojigame/PuzzlePage/EmojiPuzzleAdapter.kt [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+
+        fun bind(puzzle: Puzzle) {
+            emojiPuzzleText.text = puzzle.emojiPuzzle
+            descriptionText.text = puzzle.description
+            answerText.addTextChangedListener(getOnTextChangedListener())
+            guessButton.setOnClickListener {
+                val inputText = answerText.text.toString().trim().toLowerCase()
+
+                val isCorrect = inputText == puzzle.answer.toLowerCase()
+                feedbackText.text = if (isCorrect) "Correct!" else "Incorrect"
```
Comment (lpalac4) [inline app/src/main/java/com/example/emojigame/PuzzlePage/EmojiPuzzleAdapter.kt:53]: #nth lets remove these hardcoded string literals

File: app/.gitignore (hunk 3 above)
Comment (AlexGuler) [reply app/.gitignore:1]: Ok thanks!

File: app/build.gradle (hunk 1 above)
Comment (AlexGuler) [reply app/build.gradle:28]: Ok thank you!

File: .idea/misc.xml (hunk 2 above)
Comment (AlexGuler) [reply .idea/misc.xml:1]: Will Do

File: app/src/main/res/layout/activity_emoji_puzzle.xml (hunk 4 above)
Comment (AlexGuler) [reply app/src/main/res/layout/activity_emoji_puzzle.xml:17]: Sounds good

File: app/.gitignore (hunk 3 above)
Comment (AlexGuler) [reply app/.gitignore:1]: Will do

File: app/src/main/res/layout/activity_main.xml (hunk 5 above)
Comment (AlexGuler) [reply app/src/main/res/layout/activity_main.xml:11]: Yeah, I was thinking of doing that but then I just wanted to get something out the door.

File: app/src/main/res/layout/cell_choose_category.xml (hunk 6 above)
Comment (AlexGuler) [reply app/src/main/res/layout/cell_choose_category.xml:15]: Changed them all!

File: app/src/main/java/com/example/emojigame/MainPage/MainActivity.kt (hunk 7 above)
Comment (AlexGuler) [reply app/src/main/java/com/example/emojigame/MainPage/MainActivity.kt:33]: Changed to coroutine scope, thanks!

File: app/src/main/java/com/example/emojigame/PuzzlePage/EmojiPuzzleActivity.kt (hunk 8 above)
Comment (AlexGuler) [reply app/src/main/java/com/example/emojigame/PuzzlePage/EmojiPuzzleActivity.kt:41]: Ok

File: app/src/main/java/com/example/emojigame/PuzzlePage/EmojiPuzzleActivity.kt (hunk 9 above)
Comment (AlexGuler) [reply app/src/main/java/com/example/emojigame/PuzzlePage/EmojiPuzzleActivity.kt:32]: Sounds good!

**Pull request closed without merging**